chain-factory/tests/chain_factory_tests.rs

```diff
@@ -0,0 +1,232 @@
+use chain_config::{chain_config_proxy, StakeMultiArg};
+use chain_factory::{
+    chain_factory_proxy::{self, ContractMapArgs, ScArray},
+    common::storage::CommonStorage,
+    factory::ScArray as ContractScArray,
+};
+use multiversx_sc::types::{
+    BigUint, CodeMetadata, MultiValueEncoded, TestAddress, TestSCAddress, TokenIdentifier,
+};
+use multiversx_sc_scenario::{
+    api::StaticApi, imports::MxscPath, managed_biguint, ExpectError, ScenarioTxRun,
+    ScenarioTxWhitebox, ScenarioWorld,
+};
+
+const FACTORY_ADDRESS: TestSCAddress = TestSCAddress::new("chain-factory");
+const CODE_PATH: MxscPath = MxscPath::new("output/chain-factory.mxsc.json");
+
+const CONFIG_ADDRESS: TestSCAddress = TestSCAddress::new("chain-config");
+const CONFIG_CODE_PATH: MxscPath = MxscPath::new("../chain-config/output/chain-factory.mxsc.json");
+
+const OWNER: TestAddress = TestAddress::new("owner");
+const OWNER_BALANCE: u64 = 100_000_000_000;
+const DEPLOY_COST: u64 = 10_000_000_000;
+
+fn world() -> ScenarioWorld {
+    let mut blockchain = ScenarioWorld::new();
+
+    blockchain.register_contract(CODE_PATH, chain_factory::ContractBuilder);
+    blockchain.register_contract(CONFIG_CODE_PATH, chain_config::ContractBuilder);
+
+    blockchain
+}
+
+struct ChainFactoryTestState {
+    world: ScenarioWorld,
+}
+
+impl ChainFactoryTestState {
+    fn new(
```

Comment:
This is `chain-factory` contract. Why on `new` you are deploying `chain-config` contract?